modules/gateways/blockonomics/blockonomics.php

```diff
@@ -760,7 +760,7 @@ public function examine_server_callback_urls($response, $currency)
 
         $callback_secret = $this->getCallbackSecret();
         $callback_url = $this->getCallbackUrl($callback_secret);
-        $base_url = preg_replace('/https?:\/\//', '', substr($callback_url, 0, -48));
+        $base_url = preg_replace('/https?:\/\//', '', substr($callback_url, 0, -1 * (strlen($callback_secret) + 8)));
```

Comment:
Why is the 8 added. Please add a code comment so that this code is more understandable

Reply:
Updated the extraction logic to use `strtok` instead of manually working on it. it'll now split the string before `?` and give the callback URL without needing to use substr